### PR TITLE
Refactor DataSource

### PR DESCRIPTION
**Pull request by daolq3012**

### CODE REVIEW

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                        values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
+                        values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL,
+                                user.getAvatarUrl());
+                        values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
+                                user.getSubscriptionsUrl());
+                        database.insert(UserDbHelper.UserEntry.TABLE_NAME, null, values);
+                    }
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                    SQLiteDatabase database = mHelper.getWritableDatabase();
+                    ContentValues values = new ContentValues();
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
+                            user.getSubscriptionsUrl());
+                    database.insert(UserDbHelper.UserEntry.TABLE_NAME, null, values);
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
+                            user.getSubscriptionsUrl());
+                    database.update(UserDbHelper.UserEntry.TABLE_NAME, values,
+                            UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN + "= ?",
+                            new String[] { user.getLogin() });
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            public void subscribe(CompletableEmitter emitter) throws Exception {
+                try {
+                    SQLiteDatabase database = mHelper.getWritableDatabase();
+                    final String selection =
+                            UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN + " LIKE ?";
+                    final String[] selectionArgs = { user.getLogin() };
+                    database.delete(UserDbHelper.UserEntry.TABLE_NAME, selection, selectionArgs);
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                    ContentValues values = new ContentValues();
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
+                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
+                            user.getSubscriptionsUrl());
+                    database.insertWithOnConflict(UserDbHelper.UserEntry.TABLE_NAME, null, values,
+                            SQLiteDatabase.CONFLICT_REPLACE);
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+    public Completable deleteAllUsers() {
+        return Completable.create(new CompletableOnSubscribe() {
+            @Override
+            public void subscribe(CompletableEmitter emitter) throws Exception {
+                try {
+                    SQLiteDatabase database = mHelper.getWritableDatabase();
+                    database.delete(UserDbHelper.UserEntry.TABLE_NAME, null, null);
+                    database.close();
+                    emitter.onComplete();
+                } catch (RuntimeException e) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java]: INFO: Catching 'RuntimeException' is not allowed.

File: app/src/test/java/com/fstyle/structure_android/data/repository/UserRepositoryTest.java [hunk 7]
```diff
@@ -25,4 +72,7 @@ public void getAllUser() throws Exception {
 
     public void getUserByUserLogin() throws Exception {
     }
+
+    private void setUserAvailable(SQLiteDatabase database, User user) {
```
Comment (le-quang-dao) [inline app/src/test/java/com/fstyle/structure_android/data/repository/UserRepositoryTest.java]: WARNING: 'user' hides a field.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -24,4 +24,6 @@
     <string name="error_lenght_from_0_to_100">Limit value must be from 0 to 100</string>
     <string name="must_not_empty">Must not empty</string>
     <string name="error_unusable_characters">There are unusable characters</string>
+    <string name="a">a</string>
```
Comment (le-quang-dao) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.a` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -24,4 +24,6 @@
...
+    <string name="text_120">120</string>
```
Comment (le-quang-dao) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.text_120` appears to be unused
 Unused resources make applications larger and slow down builds.

**Comment by daolq3012** [comment]
This PR unnecessary!
Close!

**Pull request closed without merging**